### ISSUE TITLE
Add an implicit cast from Number to Currency

### ISSUE THREAD

**Issue by gcampax** [stanford-oval/thingtalk#17]
The following works:
```
alert me when bitcoin is over $8000
```

while the following does not parse correctly:
```
alert me when bitcoin is over 8000
```

This is because the latter preprocesses as a NUMBER, not a CURRENCY, and the two cannot be casted into each other. It would be nice if numeric constants could be accepted as Currency though (implicitly USD for en-US locale, EUR for it-IT etc.)

**Comment by AydanPirani** [comment]
Hello, I am currently working on this issue!

**Comment by gcampax** [comment]
Fixed by #94
